lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -204,15 +200,18 @@ def cluster_resource_discoverer
       )
     end
 
-    def ejson_provisioner
-      @ejson_provisioner ||= EjsonSecretProvisioner.new(
-        namespace: @namespace,
-        context: @context,
-        template_dir: @template_dir,
-        logger: @logger,
-        statsd_tags: @namespace_tags,
-        selector: @selector,
-      )
+    def ejson_provisioners
+      @ejson_provisoners ||= TemplateDiscovery.ejson_secret_templates(@template_paths).map do |secret_template|
+        EjsonSecretProvisioner.new(
+          namespace: @namespace,
+          context: @context,
+          ejson_keys_secret: ejson_keys_secret,
+          template_dir: File.dirname(secret_template),
```

Comment:
I think passing the ejson_keys_secret in from DeployTask is the right thing to do, but I'm actually going to revert this change and push it forward to a follow-up PR. The reason for that being it involves a number of changes to the test suite and makes also makes a change to the EjsonSecretProvisioner#new signature which requires a discussion all its own. We can accomplish this PRs goals without having this scope-creep in